### PR TITLE
Optimize nonbonded precomputed

### PR DESCRIPTION
**Pull request by badisa**
Modifies the nonbonded precomputed potential to match up with the speed optimizations made in the regular nonbonded potentials.

Improves performance of RBFE runs by about 6%

# Benchmarks
A10 card
Only focus on the RBFE tests of the benchmark as they have the nonbonded precomputed potentials

## Master
```
hif2a-rbfe: N=8840 speed: 561.40ns/day dt: 1.5fs (ran 100000 steps in 23.18s)
solvent-rbfe: N=6317 speed: 684.22ns/day dt: 1.5fs (ran 100000 steps in 19.03s)
```

## Optimized Nonbonded Precomputed
```
hif2a-rbfe: N=8840 speed: 596.80ns/day dt: 1.5fs (ran 100000 steps in 21.72s)
solvent-rbfe: N=6317 speed: 726.65ns/day dt: 1.5fs (ran 100000 steps in 17.84s)
```

### CODE REVIEW

**Review by mcwitt — APPROVED** [review]

File: tests/test_local_move.py [hunk 1]
```diff
@@ -265,5 +265,5 @@ def local_move(_, steps=500):
         return (xs[-1], boxes[-1]), None
 
     assert_no_drift(
-        (x0[-1], boxes[-1]), local_move, num_particles_near_ligand, n_local_resampling_iterations=250, threshold=0.05
+        (x0[-1], boxes[-1]), local_move, num_particles_near_ligand, n_local_resampling_iterations=250, threshold=0.055
```
Comment (mcwitt) [inline tests/test_local_move.py:268]: Why did this change? Which (if any) changes in this PR would we expect to break bitwise equivalence with the current version?

Comment (badisa) [reply tests/test_local_move.py:268]: The change that made this difference was the erfc approximation we make for float32: https://github.com/proteneer/timemachine/blob/master/timemachine/cpp/src/kernels/k_nonbonded_common.cuh#L34-L44

File: tests/test_barostat.py [hunk 2]
```diff
@@ -185,12 +182,24 @@ def test_barostat_is_deterministic():
         seed,
     )
 
-    ctxt = custom_ops.Context(coords, v_0, host_box, integrator_impl, u_impls, barostat=baro)
+    ctxt = custom_ops.Context(coords, v_0, host_box, integrator.impl(), u_impls, barostat=baro)
```
Comment (proteneer) [inline tests/test_barostat.py:185]: ooh - is this safe now? That's exciting

File: tests/test_barostat.py [hunk 3]
```diff
@@ -185,12 +182,24 @@ def test_barostat_is_deterministic():
...
+        pressure,
+        temperature,
+        group_indices,
+        barostat_interval,
+        u_impls,
+        seed,
+    )
+    ctxt = custom_ops.Context(coords, v_0, host_box, integrator.impl(), u_impls, barostat=baro)
+    ctxt.multiple_steps(15)
+    assert compute_box_volume(atm_box) == compute_box_volume(ctxt.get_box())
```
Comment (proteneer) [inline tests/test_barostat.py:202]: maybe a comment here saying what this test is doing? (eg we get bitwise reproducible boxes?)

**Review by proteneer — APPROVED** [review]
lgtm

File: tests/test_barostat.py (hunk 2 above)
Comment (badisa) [reply tests/test_barostat.py:185]: As of https://github.com/proteneer/timemachine/commit/5a1ddfaba3e396a71240d3dea10c55b0813f47ea we don't have to manage wrapped objects the way we use to.

File: tests/test_barostat.py (hunk 3 above)
Comment (badisa) [reply tests/test_barostat.py:203]: Added comment: https://github.com/proteneer/timemachine/pull/1023/commits/6a80a10cbfec03dc37a7655bf121914aef702df2

**Pull request merged**